src/compiler/scala/tools/nsc/typechecker/Unapplies.scala

```diff
@@ -128,7 +128,11 @@ trait Unapplies extends ast.TreeDSL {
    */
   def factoryMeth(mods: Modifiers, name: TermName, cdef: ClassDef): DefDef = {
     val tparams   = constrTparamsInvariant(cdef)
-    val cparamss  = constrParamss(cdef)
+    // don't propagate e.g. @volatile annot to apply's argument
+    def valDefWithoutAnnots(vd: ValDef): ValDef =
+      treeCopy.ValDef(vd, vd.mods.mapAnnotations(_ => Nil), vd.name, vd.tpt, vd.rhs)
```

Comment:
One example of an annotation that we might want to copy to the parameter is `@deprecatedName`. Failure to do so would regress the following:

```
scala> case class A(@deprecatedName('oldName) newName: Any)
defined class A

scala> new A(oldName = "methuselah")
warning: there was one deprecation warning; re-run with -deprecation for details
res24: A = A(methuselah)

scala> A(oldName = "methuselah")
warning: there was one deprecation warning; re-run with -deprecation for details
res25: A = A(methuselah)
```